src/mlpack/methods/bayesian_linear_regression/bayesian_linear_regression.cpp

```diff
@@ -0,0 +1,175 @@
+    gamma = sum(eigVal / (alpha / beta + eigVal));
+    alpha = gamma / dot(omega, omega);
+
+    // Update beta.
+    const arma::rowvec temp = t - omega.t() * phi;
+    beta = (data.n_cols - gamma) / dot(temp, temp);
+
+    // Compute the stopping criterion.
+    deltaAlpha += alpha;
+    deltaBeta += beta;
+    crit = std::abs(deltaAlpha / alpha + deltaBeta / beta);
+    i++;
+  }
+  // Compute the covariance matrix for the uncertainties later.
+  matCovariance = eigVec * diagmat(1 / (beta * eigVal + alpha)) * eigVecInv;
+
+  Timer::Stop("bayesian_linear_regression");
+
+  return RMSE(data, responses);
+}
+
+void BayesianLinearRegression::Predict(const arma::mat& points,
+                                       arma::rowvec& predictions) const
+{
+  // Center and scale the points before applying the model.
+  arma::mat matX;
+  CenterScaleDataPred(points, matX);
+  predictions = omega.t() * matX + responsesOffset;
+}
+
+void BayesianLinearRegression::Predict(const arma::mat& points,
+                                       arma::rowvec& predictions,
+                                       arma::rowvec& std) const
+{
+  // Center and scale the points before applying the model.
+  arma::mat matX;
+  CenterScaleDataPred(points, matX);
+  predictions = omega.t() * matX + responsesOffset;
+  // Compute the standard deviation dor each points.
+  std = sqrt(Variance() + sum((matX % (matCovariance * matX)), 0));
+}
+
+double BayesianLinearRegression::RMSE(const arma::mat& data,
+                                      const arma::rowvec& responses) const
+{
+  arma::rowvec predictions;
+  Predict(data, predictions);
+  return sqrt(mean(square(responses - predictions)));
+}
+
+double BayesianLinearRegression::CenterScaleData(const arma::mat& data,
+                                                 const arma::rowvec& responses,
+                                                 arma::mat& dataProc,
+                                                 arma::rowvec& responsesProc)
+{
+  // Initialize the offsets to their neutral forms.
+  responsesOffset = 0.0;
+  if (!centerData && !scaleData)
+  {
+    dataProc = data;
+    responsesProc = responses;
+  }
+
+  else if (centerData && !scaleData)
+  {
+    dataOffset = mean(data, 1);
+    responsesOffset = mean(responses);
+    dataProc = data.each_col() - dataOffset;
+    responsesProc = responses - responsesOffset;
+  }
+
+  else if (!centerData && scaleData)
+  {
+    dataScale = stddev(data, 0, 1);
+    dataProc = data.each_col() / dataScale;
+  }
+
+  else
+  {
+    dataOffset = mean(data, 1);
+    dataScale = stddev(data, 0, 1);
+    responsesOffset = mean(responses);
+    dataProc = (data.each_col() - dataOffset).each_col() / dataScale;
+    responsesProc = responses - responsesOffset;
+  }
+  return responsesOffset;
+}
+
+void BayesianLinearRegression::CenterScaleDataPred(
+    const arma::mat& data,
+    arma::mat& dataProc) const
+{
+  if (!centerData && !scaleData)
+  dataProc = data;
```

Comment:
```suggestion
    dataProc = data;
```
Also, we could use the same alias trick here. :+1: